src/System.Diagnostics.Context/SpanBuilder.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System.Collections.Generic;
```

Comment:
I know that the Java implementation used a explicit builder, but I think we can live without it.   We can make Span be its own builder.    It can have 'With*' methods and a 'Start' (rather than Build()) method, and it achieves the same result with less code (and more efficiently).